### PR TITLE
feat(argo-cd): Enable to add certificate to ApplicationSet Controller

### PR DESCRIPTION
**Pull request by yu-croco**
Resolves https://github.com/argoproj/argo-helm/issues/2028 🙋 

---

Signed-off-by: yu-croco <yu.croco@gmail.com>

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -849,6 +849,45 @@ controller:
     # -- List of custom rules for the application controller's ClusterRole resource
     rules: []
 
+  # TLS certificate configuration via cert-manager
+  ## Ref: https://argo-cd.readthedocs.io/en/stable/operator-manual/tls/#tls-configuration
+  certificate:
```
Comment (yu-croco) [inline charts/argo-cd/values.yaml:854]: same as [server's definition](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L1696-L1730) 🙋

File: charts/argo-cd/templates/argocd-application-controller/certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{{- if .Values.controller.certificate.enabled -}}
```
Comment (yu-croco) [inline charts/argo-cd/templates/argocd-application-controller/certificate.yaml:1]: same as https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-server/certificate.yaml 🙋

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Hi @yu-croco
You implemented it in the "wrong" controller.

The issue wants that feature on the ApplicationSET controller not the normal controller :-)

**Comment by yu-croco** [comment]
Hi @mkilchhofer , thank you for your review.
I fixed on https://github.com/argoproj/argo-helm/pull/2052/commits/8f8049499e5ce22a154444a940fca2fb5fa24213 🙋

**Review by mbevc1 — APPROVED** [review]

**Review by jmeridth — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**